### ISSUE TITLE
[Bug]: browserSupportsCspV3() returns false which is bad

### ISSUE THREAD

**Issue by the-djmaze** [nextcloud/server#35013]
### ⚠️ This issue respects the following points: ⚠️

- [X] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [X] This issue is **not** already reported on Github _(I've searched it)_.
- [X] Nextcloud Server **is** up to date. See [Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) for supported versions.
- [X] Nextcloud Server **is** running on 64bit capable CPU, PHP and OS.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Bug description

[browserSupportsCspV3()](https://github.com/nextcloud/server/blob/0dabc6abc6fe203b5126a6f4fc33ddb55cfd41ea/lib/private/Security/CSP/ContentSecurityPolicyNonceManager.php#L80) should always return TRUE and never do browser sniffing in any way.
It's terrible and indeed now bites again.

My Nextcloud app fails in Microsoft Edge AND iOS while both work fine with CSP Level 3 (just like many other browsers and TOR, but you return false).

Because browserSupportsCspV3() returns `false`, the `nonce` isn't working.

Just remove the check as CSP Level 1 & 2 browsers are ancient.

### Expected behavior

browserSupportsCspV3() always return true

### Additional info

https://github.com/nextcloud/server/blob/0dabc6abc6fe203b5126a6f4fc33ddb55cfd41ea/lib/private/Security/CSP/ContentSecurityPolicyNonceManager.php#L80

iOS 16 https://github.com/the-djmaze/snappymail/issues/631
Microsoft Edge https://github.com/the-djmaze/snappymail/issues/633
firefox-esr https://github.com/the-djmaze/snappymail/issues/634

https://caniuse.com/?search=Nonce
https://caniuse.com/?search=strict-dynamic

### TL;DR
Here a good read about `strict-dynamic`
https://content-security-policy.com/strict-dynamic/

And being an annoying evangelist :)
https://www.sitepoint.com/why-browser-sniffing-stinks/
https://stackoverflow.com/questions/661213/why-is-browser-sniffing-not-a-recommended-practice
https://css-tricks.com/browser-detection-is-bad/

### Request::USER_AGENT_*

There are probably more locations where these are used, but i didn't check them.

**Comment by joshtrichards** [comment]
Related: #10207 

This behavior has since been changed: #44412
